### ISSUE TITLE
plot methods for perf and tune to use ggplot2

### ISSUE THREAD

**Issue by aljabadi** [mixOmicsTeam/mixOmics#112]
scrap base plots and include customisation args (pch, pch.size, col, cex)

**Comment by aljabadi** [comment]
``` r
suppressMessages(devtools::load_all('/Users/alabadi/Projects/dev/R/_work/mixOmics/mixOmics_ajabadi/mixOmics_ajabadi'))
data(liver.toxicity)
X <- liver.toxicity$gene
Y <- liver.toxicity$clinic
# for pls1
y <- liver.toxicity$clinic[, "ALB.g.dL."]
folds <- 5
tune.pls1.liver <- pls(X = X, Y = y, ncomp = 4, mode = 'regression')
tune.pls2.liver <- pls(X = X, Y = Y, ncomp = 4, mode = 'regression')
```

# PLS1

### nrepeat \> 2

``` r
set.seed(33)
nrep <- 3
Q2.pls1.liver <- perf(tune.pls1.liver, validation = 'Mfold', 
                      folds = folds, nrepeat = nrep)
plot(Q2.pls1.liver, criterion = 'RMSEP', sd = TRUE)
```

![](https://i.imgur.com/FkdgsRm.png)

``` r
plot(Q2.pls1.liver, criterion = 'Q2', sd = TRUE)
```

![](https://i.imgur.com/CCwaUQB.png)

``` r
plot(Q2.pls1.liver, criterion = 'Q2.total', sd = TRUE)
```

![](https://i.imgur.com/WmBYHgE.png)

``` r
plot(Q2.pls1.liver, criterion = 'cor.tpred', sd = TRUE)
#> Error in plot.perf.pls1.mthd(Q2.pls1.liver, criterion = "cor.tpred", sd = TRUE): 'criterion' must be one of names(Q2.pls1.liver$measures): 
#> MSEP, RMSEP, R2, Q2, Q2.total, RSS, PRESS
```

### nrepeat \<= 2

``` r
set.seed(33)
nrep <- 1
Q2.pls1.liver.rep1 <- perf(tune.pls1.liver, validation = 'Mfold', 
                      folds = folds, nrepeat = nrep)
plot(Q2.pls1.liver.rep1, criterion = 'RMSEP', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/VQRxEk0.png)

``` r
plot(Q2.pls1.liver.rep1, criterion = 'Q2', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/1ok8Sc4.png)

``` r
plot(Q2.pls1.liver.rep1, criterion = 'Q2.total', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/ZiFJySx.png)

``` r
plot(Q2.pls1.liver, criterion = 'cor.tpred', sd = TRUE) ## error
#> Error in plot.perf.pls1.mthd(Q2.pls1.liver, criterion = "cor.tpred", sd = TRUE): 'criterion' must be one of names(Q2.pls1.liver$measures): 
#> MSEP, RMSEP, R2, Q2, Q2.total, RSS, PRESS
```

# PLS2

### nrepeat \> 2

``` r
set.seed(33)
nrep <- 3
Q2.pls2.liver <- perf(tune.pls2.liver, validation = 'Mfold', 
                      folds = folds, nrepeat = nrep)
# debugonce(plot)
plot(Q2.pls2.liver, criterion = 'RMSEP', sd = TRUE)
```

![](https://i.imgur.com/A72Vom7.png)

``` r
plot(Q2.pls2.liver, criterion = 'Q2', sd = TRUE, cex = 1.3)
```

![](https://i.imgur.com/SgAvYJG.png)

``` r
plot(Q2.pls2.liver, criterion = 'Q2.total', sd = TRUE) 
```

![](https://i.imgur.com/1sUQsum.png)

``` r
plot(Q2.pls2.liver, criterion = 'cor.tpred', sd = TRUE)
```

![](https://i.imgur.com/ywELIAA.png)

### nrepeat \<= 2

``` r
set.seed(33)
nrep <- 1
Q2.pls2.liver.rep1 <- perf(tune.pls2.liver, validation = 'Mfold', 
                      folds = folds, nrepeat = nrep)
plot(Q2.pls2.liver.rep1, criterion = 'RMSEP', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/MbA12jb.png)

``` r
plot(Q2.pls2.liver.rep1, criterion = 'Q2', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/QxIziic.png)

``` r
plot(Q2.pls2.liver.rep1, criterion = 'Q2.total', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/fRGJLbG.png)

``` r
plot(Q2.pls2.liver.rep1, criterion = 'cor.tpred', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/SC7VSWU.png)

# plot args

``` r
plot(Q2.pls2.liver.rep1, criterion = 'Q2.total', sd = TRUE)
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/JyE2sEc.png)

``` r
plot(Q2.pls2.liver.rep1, criterion = 'Q2.total', sd = TRUE, pch = 16, pch.size = 3, col = color.mixo(2))
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/uQsoof9.png)

### other customisation examples

``` r
plot(Q2.pls2.liver.rep1, criterion = 'Q2.total', sd = TRUE) +
    geom_text(x = 1, y = 0.0975, label = 'Q2 = 0.0975', hjust=-0.5, vjust=-0.5) 
#> error bars cannot be calculated as nrepeat < 3
```

![](https://i.imgur.com/mkNJOyJ.png)

**Comment by aljabadi** [comment]
fixed as of >= 6.14.1
